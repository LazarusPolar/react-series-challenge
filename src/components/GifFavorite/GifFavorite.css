.GifFavorite {
    z-index: 10px;
    display: inline-block;
}

.GifFavorite img {
    width: 400px;
    height: 250px;
    z-index: 10px;
    display: inline-block;
}