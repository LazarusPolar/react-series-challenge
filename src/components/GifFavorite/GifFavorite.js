import React, { Component } from 'react';
import SearchBar from '../../UI/SearchBar';
import GifCards from '../GifCards/GifCards';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import SearchFunctions from '../../actions/searchValues';

class GifFavorite extends Component {
    constructor(props) {
        super(props);
        this.state = {
            searchedValue: null, 
            searchResults: props.favoritedImages, 
        }
    }

    updateState = (event) => {
        console.log(event);
        this.setState({ searchedValue: event })
    }

    handleSearch = () => {
        console.log(this.props.favoritedImages);
        //const searchParam = event.target.value;
        let newArray = Object.keys(this.props.favoritedImages).map((key, index) => {
            return this.props.favoritedImages[index]
        }).filter(word => {
            console.log("WORD", word);
            console.log("SEARCHEDVALUE", this.state.searchedValue)
            return word.searchedValue.includes(this.state.searchedValue)});
        
        console.log("NUEVO ARRAY", newArray);

        if(newArray !== undefined && newArray !== null && newArray.length >= 0){
            console.log("Entro a actualizar");
            this.setState({ searchResults: newArray });
        } 
        
    }

    render() {
        return (
            <div className = "GifFavorite">
                <SearchBar 
                updateState = { this.updateState } 
                handleSearch = { this.handleSearch } />
    
                <GifCards 
                gifData = { this.state.searchResults } 
                searchedGifs = { this.props.searchedFavoriteGifs } 
                searchedValue = { this.props.searchedValue }/>
            </div>
        );
    }

}

const mapStateToProps = (state) => {
    console.log(state);
    const { greeting, favoritedImages } = state;
    return {
        greeting, favoritedImages
    }
}

const mapDispatchToProps = (dispatch) => {
    const { searchedFavoriteGifs, searchedTrendingGifs, searchedSpecifiedGifs, updateSearchValue } = SearchFunctions.creators;
    return bindActionCreators( { searchedFavoriteGifs, searchedTrendingGifs, searchedSpecifiedGifs, updateSearchValue }, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(GifFavorite);