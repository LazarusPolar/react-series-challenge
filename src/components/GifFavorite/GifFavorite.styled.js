import styled from 'styled-components';

export const StyledGifFavorite = styled.div`
    text-align: center;
    padding-left: 0px;
    padding-right: 2px;
    width: 100%;
    background-color: ${props => props.theme.cardBackground};
`

