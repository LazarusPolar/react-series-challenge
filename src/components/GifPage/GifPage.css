.GifPage {
    z-index: 10px;
    display: inline-block;
}

.GifPage GifCards {
    width: 400px;
    height: 250px;
    z-index: 10px;
    display: inline-block;
}