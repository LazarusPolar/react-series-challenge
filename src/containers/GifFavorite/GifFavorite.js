import React, { Component } from 'react';

class GifFavorite extends Component {

}

export default GifFavorite;