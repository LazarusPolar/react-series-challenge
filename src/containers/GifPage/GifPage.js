import React, {Component} from 'react'

class GifPage extends Component {
    constructor(props){
        super(props);
        this.state = {
            variable: 1
        };
    }

    render() {
        return(
        <div>
            <header>
                <nav>
                    <ul>
                        <li>Home</li>
                        <li>Favorites</li>
                    </ul>
                </nav>
            </header>

            <div>
                <input type="text" placeholder = "Search for awesome gifs" />
            </div>
        </div>)
    }
}

export default GifPage;