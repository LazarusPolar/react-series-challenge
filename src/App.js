import React, { Component } from 'react';
import { Provider } from 'react-redux';
import store from './store/index';
import GifPage from './components/GifPage/GifPage'
import GifFavorite from './components/GifFavorite/GifFavorite';
import { GifSection, StyledHeader, StyledNavLink, StyledUl, StyledLi, StyledDiv } from './App.styled'
import { Route, Switch, Redirect } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import { themeSunny, themeNightly } from './globalInjection'
import ThemeSelector from './components/ThemeSelector/ThemeSelector'

class App extends Component {

  constructor(props) {
    super(props);
    this.state = {
      nightMode: true,
    }
  }

  changeTheme = () => {
    this.setState ( (prevState) => ({ nightMode: !prevState.nightMode }) );
    console.log("Estado es: " + this.state.nightMode);
  }

  render() {
    return (
      <ThemeProvider theme = { this.state.nightMode ? themeSunny : themeNightly }>
        <Provider store = {store}>
        <StyledDiv>   
              <StyledHeader {...this.props}>
                <StyledUl>
                    <StyledLi><StyledNavLink 
                    to = "/home"
                    activeClassName = "act"
                    exact>
                    Home </StyledNavLink></StyledLi>

                    <StyledLi><StyledNavLink 
                    to = "/favorite"
                    activeClassName = "act"
                    exact>
                    Favorites</StyledNavLink></StyledLi>
                </StyledUl>
                <ThemeSelector changeTheme = {this.changeTheme}/>
              </StyledHeader>
              <GifSection>
                <Switch>
                  <Route path = "/home" exact component = {GifPage} />
                  <Route path = "/favorite" exact component = {GifFavorite} />
                  <Redirect from = "/" to = "/home" />
                </Switch>
              </GifSection>
        </StyledDiv>
        </Provider>
      </ThemeProvider>
    );
  }
}

export default App;
