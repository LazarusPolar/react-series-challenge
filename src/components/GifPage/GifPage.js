import React, {Component} from 'react';
import GifCards from '../GifCards/GifCards'
import './GifPage.css';
import SearchBar from '../../UI/SearchBar';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import SearchFunctions from '../../actions/searchValues';
import ErrorBoundary from '../../ErrorBoundary/ErrorBoundary';

class GifPage extends Component {

    componentDidMount() {      
        this.props.requestApiData();
    }

    handleSearch = () => {
        this.props.requestApiDataSearched();
    }

    updateState = (event) => {
        this.props.updateSearchValue({searchValue: event});
    }

    render() {
        let newArray = [];
        let newObject = {};

        if(this.props.gifs !== undefined ){
            for(let a = 0; a <= Object.keys(this.props.gifs).length - 1; a++){
                newObject = {};
                newObject.id = this.props.gifs[a].id;
                newObject.url = this.props.gifs[a].images.downsized.url;
                newArray.push(newObject);
            }
        }

        return(
            <div className = "GifPage">
                <SearchBar updateState = { this.updateState } handleSearch = {this.handleSearch} />
                <ErrorBoundary>
                <GifCards gifData = { newArray } addRemoveGif = {this.props.addRemoveGifFavorites} searchedValue = {this.props.searchedValue}/> 
                </ErrorBoundary>        
            </div>)
    }
}

const mapStateToProps = (state) => {
    const { gifs, searchedValue } = state;
    return {
        gifs, searchedValue
    }
}

const mapDispatchToProps = (dispatch) => {
    const { addRemoveGifFavorites, updateSearchValue, requestApiData, requestApiDataSearched } = SearchFunctions.creators;
    return bindActionCreators( { addRemoveGifFavorites, updateSearchValue, requestApiData, requestApiDataSearched }, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(GifPage);