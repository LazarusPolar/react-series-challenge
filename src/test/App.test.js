import App from '../App';
import React from 'react';
import { shallow, mount } from 'enzyme';
import { Route } from 'react-router-dom';
import GifPage from '../components/GifPage/GifPage'
import GifFavorite from '../components/GifFavorite/GifFavorite'
import {StyledLi, StyledNavLink} from '../App.styled'

describe('App', () => {
  let wrapper;
  let func = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <App />
    );
  });

  test('should set the routes ', () => {
    expect(
      wrapper.contains(<Route path = "/home" exact component = {GifPage} />)
    ).toBe(true);
  });

  test('should set the routes ', () => {
    expect(
      wrapper.contains(<Route path = "/home" exact component = {GifPage} />)
    ).toBe(true);
  });

  test('should set the routess ', () => {
    expect(
      wrapper.contains(<Route path = "/favorite" exact component = {GifFavorite} />)
    ).toBe(true);
  });

  test('should set nav links ', () => {
    expect(
      wrapper.contains(<StyledLi><StyledNavLink 
        to = "/home"
        exact>
        Home </StyledNavLink></StyledLi>)
    ).toBe(true);
  });

  test('should set nav links ', () => {
    expect(
      wrapper.contains(<StyledLi><StyledNavLink 
        to = "/favorite"
        exact>
        Favorites</StyledNavLink></StyledLi>)
    ).toBe(true);
  });
  
})
