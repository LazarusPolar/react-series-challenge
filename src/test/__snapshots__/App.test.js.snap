// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Provider
        store={
                Object {
                        "dispatch": [Function],
                        "getState": [Function],
                        "replaceReducer": [Function],
                        "subscribe": [Function],
                        Symbol(observable): [Function],
                      }
        }
>
        <styled.div>
                <styled.div>
                        <styled.ul>
                                <styled.li>
                                        <Styled(NavLink)
                                                activeClassName="act"
                                                exact={true}
                                                to="/home"
                                        >
                                                Home 
                                        </Styled(NavLink)>
                                </styled.li>
                                <styled.li>
                                        <Styled(NavLink)
                                                activeClassName="act"
                                                exact={true}
                                                to="/favorite"
                                        >
                                                Favorites
                                        </Styled(NavLink)>
                                </styled.li>
                        </styled.ul>
                        <ThemeSelector
                                changeTheme={[Function]}
                        />
                </styled.div>
                <styled.div>
                        <Switch>
                                <Route
                                        component={[Function]}
                                        exact={true}
                                        path="/home"
                                />
                                <Route
                                        component={[Function]}
                                        exact={true}
                                        path="/favorite"
                                />
                                <Redirect
                                        from="/"
                                        push={false}
                                        to="/home"
                                />
                        </Switch>
                </styled.div>
        </styled.div>
</Provider>,
      "theme": Object {
        "cardBackground": "#fec492",
        "favoritePageBackground": "#c8bed7",
        "gifPageBackground": "#969ec3",
        "headerBackground": "#fde1cc",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <styled.div>
          <styled.div>
                    <styled.ul>
                              <styled.li>
                                        <Styled(NavLink)
                                                  activeClassName="act"
                                                  exact={true}
                                                  to="/home"
                                        >
                                                  Home 
                                        </Styled(NavLink)>
                              </styled.li>
                              <styled.li>
                                        <Styled(NavLink)
                                                  activeClassName="act"
                                                  exact={true}
                                                  to="/favorite"
                                        >
                                                  Favorites
                                        </Styled(NavLink)>
                              </styled.li>
                    </styled.ul>
                    <ThemeSelector
                              changeTheme={[Function]}
                    />
          </styled.div>
          <styled.div>
                    <Switch>
                              <Route
                                        component={[Function]}
                                        exact={true}
                                        path="/home"
                              />
                              <Route
                                        component={[Function]}
                                        exact={true}
                                        path="/favorite"
                              />
                              <Redirect
                                        from="/"
                                        push={false}
                                        to="/home"
                              />
                    </Switch>
          </styled.div>
</styled.div>,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.div>
              <styled.ul>
                            <styled.li>
                                          <Styled(NavLink)
                                                        activeClassName="act"
                                                        exact={true}
                                                        to="/home"
                                          >
                                                        Home 
                                          </Styled(NavLink)>
                            </styled.li>
                            <styled.li>
                                          <Styled(NavLink)
                                                        activeClassName="act"
                                                        exact={true}
                                                        to="/favorite"
                                          >
                                                        Favorites
                                          </Styled(NavLink)>
                            </styled.li>
              </styled.ul>
              <ThemeSelector
                            changeTheme={[Function]}
              />
</styled.div>,
            <styled.div>
              <Switch>
                            <Route
                                          component={[Function]}
                                          exact={true}
                                          path="/home"
                            />
                            <Route
                                          component={[Function]}
                                          exact={true}
                                          path="/favorite"
                            />
                            <Redirect
                                          from="/"
                                          push={false}
                                          to="/home"
                            />
              </Switch>
</styled.div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <styled.ul>
                  <styled.li>
                                    <Styled(NavLink)
                                                      activeClassName="act"
                                                      exact={true}
                                                      to="/home"
                                    >
                                                      Home 
                                    </Styled(NavLink)>
                  </styled.li>
                  <styled.li>
                                    <Styled(NavLink)
                                                      activeClassName="act"
                                                      exact={true}
                                                      to="/favorite"
                                    >
                                                      Favorites
                                    </Styled(NavLink)>
                  </styled.li>
</styled.ul>,
                <ThemeSelector
                  changeTheme={[Function]}
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <styled.li>
                      <Styled(NavLink)
                                            activeClassName="act"
                                            exact={true}
                                            to="/home"
                      >
                                            Home 
                      </Styled(NavLink)>
</styled.li>,
                    <styled.li>
                      <Styled(NavLink)
                                            activeClassName="act"
                                            exact={true}
                                            to="/favorite"
                      >
                                            Favorites
                      </Styled(NavLink)>
</styled.li>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <Styled(NavLink)
                        activeClassName="act"
                        exact={true}
                        to="/home"
>
                        Home 
</Styled(NavLink)>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "activeClassName": "act",
                        "children": "Home ",
                        "exact": true,
                        "to": "/home",
                      },
                      "ref": null,
                      "rendered": "Home ",
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <Styled(NavLink)
                        activeClassName="act"
                        exact={true}
                        to="/favorite"
>
                        Favorites
</Styled(NavLink)>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "activeClassName": "act",
                        "children": "Favorites",
                        "exact": true,
                        "to": "/favorite",
                      },
                      "ref": null,
                      "rendered": "Favorites",
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "changeTheme": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Switch>
                <Route
                                component={[Function]}
                                exact={true}
                                path="/home"
                />
                <Route
                                component={[Function]}
                                exact={true}
                                path="/favorite"
                />
                <Redirect
                                from="/"
                                push={false}
                                to="/home"
                />
</Switch>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/home"
/>,
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/favorite"
/>,
                  <Redirect
                    from="/"
                    push={false}
                    to="/home"
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/home",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/favorite",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "from": "/",
                    "push": false,
                    "to": "/home",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Provider
          store={
                    Object {
                              "dispatch": [Function],
                              "getState": [Function],
                              "replaceReducer": [Function],
                              "subscribe": [Function],
                              Symbol(observable): [Function],
                            }
          }
>
          <styled.div>
                    <styled.div>
                              <styled.ul>
                                        <styled.li>
                                                  <Styled(NavLink)
                                                            activeClassName="act"
                                                            exact={true}
                                                            to="/home"
                                                  >
                                                            Home 
                                                  </Styled(NavLink)>
                                        </styled.li>
                                        <styled.li>
                                                  <Styled(NavLink)
                                                            activeClassName="act"
                                                            exact={true}
                                                            to="/favorite"
                                                  >
                                                            Favorites
                                                  </Styled(NavLink)>
                                        </styled.li>
                              </styled.ul>
                              <ThemeSelector
                                        changeTheme={[Function]}
                              />
                    </styled.div>
                    <styled.div>
                              <Switch>
                                        <Route
                                                  component={[Function]}
                                                  exact={true}
                                                  path="/home"
                                        />
                                        <Route
                                                  component={[Function]}
                                                  exact={true}
                                                  path="/favorite"
                                        />
                                        <Redirect
                                                  from="/"
                                                  push={false}
                                                  to="/home"
                                        />
                              </Switch>
                    </styled.div>
          </styled.div>
</Provider>,
        "theme": Object {
          "cardBackground": "#fec492",
          "favoritePageBackground": "#c8bed7",
          "gifPageBackground": "#969ec3",
          "headerBackground": "#fde1cc",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <styled.div>
            <styled.div>
                        <styled.ul>
                                    <styled.li>
                                                <Styled(NavLink)
                                                            activeClassName="act"
                                                            exact={true}
                                                            to="/home"
                                                >
                                                            Home 
                                                </Styled(NavLink)>
                                    </styled.li>
                                    <styled.li>
                                                <Styled(NavLink)
                                                            activeClassName="act"
                                                            exact={true}
                                                            to="/favorite"
                                                >
                                                            Favorites
                                                </Styled(NavLink)>
                                    </styled.li>
                        </styled.ul>
                        <ThemeSelector
                                    changeTheme={[Function]}
                        />
            </styled.div>
            <styled.div>
                        <Switch>
                                    <Route
                                                component={[Function]}
                                                exact={true}
                                                path="/home"
                                    />
                                    <Route
                                                component={[Function]}
                                                exact={true}
                                                path="/favorite"
                                    />
                                    <Redirect
                                                from="/"
                                                push={false}
                                                to="/home"
                                    />
                        </Switch>
            </styled.div>
</styled.div>,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.div>
                <styled.ul>
                                <styled.li>
                                                <Styled(NavLink)
                                                                activeClassName="act"
                                                                exact={true}
                                                                to="/home"
                                                >
                                                                Home 
                                                </Styled(NavLink)>
                                </styled.li>
                                <styled.li>
                                                <Styled(NavLink)
                                                                activeClassName="act"
                                                                exact={true}
                                                                to="/favorite"
                                                >
                                                                Favorites
                                                </Styled(NavLink)>
                                </styled.li>
                </styled.ul>
                <ThemeSelector
                                changeTheme={[Function]}
                />
</styled.div>,
              <styled.div>
                <Switch>
                                <Route
                                                component={[Function]}
                                                exact={true}
                                                path="/home"
                                />
                                <Route
                                                component={[Function]}
                                                exact={true}
                                                path="/favorite"
                                />
                                <Redirect
                                                from="/"
                                                push={false}
                                                to="/home"
                                />
                </Switch>
</styled.div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <styled.ul>
                    <styled.li>
                                        <Styled(NavLink)
                                                            activeClassName="act"
                                                            exact={true}
                                                            to="/home"
                                        >
                                                            Home 
                                        </Styled(NavLink)>
                    </styled.li>
                    <styled.li>
                                        <Styled(NavLink)
                                                            activeClassName="act"
                                                            exact={true}
                                                            to="/favorite"
                                        >
                                                            Favorites
                                        </Styled(NavLink)>
                    </styled.li>
</styled.ul>,
                  <ThemeSelector
                    changeTheme={[Function]}
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <styled.li>
                        <Styled(NavLink)
                                                activeClassName="act"
                                                exact={true}
                                                to="/home"
                        >
                                                Home 
                        </Styled(NavLink)>
</styled.li>,
                      <styled.li>
                        <Styled(NavLink)
                                                activeClassName="act"
                                                exact={true}
                                                to="/favorite"
                        >
                                                Favorites
                        </Styled(NavLink)>
</styled.li>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": <Styled(NavLink)
                          activeClassName="act"
                          exact={true}
                          to="/home"
>
                          Home 
</Styled(NavLink)>,
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "activeClassName": "act",
                          "children": "Home ",
                          "exact": true,
                          "to": "/home",
                        },
                        "ref": null,
                        "rendered": "Home ",
                        "type": [Function],
                      },
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": <Styled(NavLink)
                          activeClassName="act"
                          exact={true}
                          to="/favorite"
>
                          Favorites
</Styled(NavLink)>,
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "activeClassName": "act",
                          "children": "Favorites",
                          "exact": true,
                          "to": "/favorite",
                        },
                        "ref": null,
                        "rendered": "Favorites",
                        "type": [Function],
                      },
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "changeTheme": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Switch>
                  <Route
                                    component={[Function]}
                                    exact={true}
                                    path="/home"
                  />
                  <Route
                                    component={[Function]}
                                    exact={true}
                                    path="/favorite"
                  />
                  <Redirect
                                    from="/"
                                    push={false}
                                    to="/home"
                  />
</Switch>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <Route
                      component={[Function]}
                      exact={true}
                      path="/home"
/>,
                    <Route
                      component={[Function]}
                      exact={true}
                      path="/favorite"
/>,
                    <Redirect
                      from="/"
                      push={false}
                      to="/home"
/>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "component": [Function],
                      "exact": true,
                      "path": "/home",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "component": [Function],
                      "exact": true,
                      "path": "/favorite",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "from": "/",
                      "push": false,
                      "to": "/home",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
